libs/partners/databricks/langchain_databricks/vectorstores.py

```diff
@@ -0,0 +1,831 @@
+
+    Delete Documents:
+        .. code-block:: python
+            vector_store.delete(ids=["3"])
+
+        .. note::
+
+            The `delete` method is only supported for direct-access index.
+
+    Search:
+        .. code-block:: python
+            results = vector_store.similarity_search(query="thud",k=1)
+            for doc in results:
+                print(f"* {doc.page_content} [{doc.metadata}]")
+        .. code-block:: python
+            * thud [{'id': '2'}]
+
+        .. note:
+
+            By default, similarity search only returns the primary key and text column.
+            If you want to retrieve the custom metadata associated with the document,
+            pass the additional columns in the `columns` parameter when initializing the vector store.
+
+            .. code-block:: python
+
+                vector_store = DatabricksVectorSearch(
+                    endpoint="<your-endpoint-name>",
+                    index_name="<your-index-name>",
+                    columns=["baz", "bar"],
+                )
+
+                vector_store.similarity_search(query="thud",k=1)
+                # Output: * thud [{'bar': 'baz', 'baz': None, 'id': '2'}]
+
+    Search with filter:
+        .. code-block:: python
+            results = vector_store.similarity_search(query="thud",k=1,filter={"bar": "baz"})
+            for doc in results:
+                print(f"* {doc.page_content} [{doc.metadata}]")
+        .. code-block:: python
+            * thud [{'id': '2'}]
+
+    Search with score:
+        .. code-block:: python
+            results = vector_store.similarity_search_with_score(query="qux",k=1)
+            for doc, score in results:
+                print(f"* [SIM={score:3f}] {doc.page_content} [{doc.metadata}]")
+        .. code-block:: python
+            * [SIM=0.748804] foo [{'id': '1'}]
+
+    Async:
+        .. code-block:: python
+            # add documents
+            await vector_store.aadd_documents(documents=documents, ids=ids)
+            # delete documents
+            await vector_store.adelete(ids=["3"])
+            # search
+            results = vector_store.asimilarity_search(query="thud",k=1)
+            # search with score
+            results = await vector_store.asimilarity_search_with_score(query="qux",k=1)
+            for doc,score in results:
+                print(f"* [SIM={score:3f}] {doc.page_content} [{doc.metadata}]")
+        .. code-block:: python
+            * [SIM=0.748807] foo [{'id': '1'}]
+
+    Use as Retriever:
+        .. code-block:: python
+            retriever = vector_store.as_retriever(
+                search_type="mmr",
+                search_kwargs={"k": 1, "fetch_k": 2, "lambda_mult": 0.5},
+            )
+            retriever.invoke("thud")
+        .. code-block:: python
+            [Document(metadata={'id': '2'}, page_content='thud')]
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        endpoint: str,
+        index_name: str,
```

Comment:
Updated the constructor to take `endpoint` and `index_name` strings, instead of a `VectorSearchIndex` object ([previous code](https://github.com/langchain-ai/langchain/blob/e958f76160c6542072c7bd334fbb3c21d327e8bc/libs/community/langchain_community/vectorstores/databricks_vector_search.py#L119)). Two motivations for this change:
1. Convenience, users don't need to create client instance and query for the index.
2. Migration to Databricks SDK. The index object returned from Databricks SDK is different from `VectorSearchIndex` object. In the new signature, we can do the migration without asking any change in users' code.